app/controllers/cookies_controller.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+class CookiesController < ApplicationController
+  before_action :set_cookie_form, only: :show
+
+  def show; end
+
+  def update
+    analytics_consent = params[:cookies_form][:analytics_consent]
+    if %w[on off].include?(analytics_consent)
+      cookies[:cookie_consent_1] = { value: analytics_consent, expires: 1.year }
```

Comment:
Should expires value have specific date when it does expire. For example
`1.year.from_now`

Reply:
I got this from the docs: https://api.rubyonrails.org/classes/ActionDispatch/Cookies.html

Other examples I've just found online do specify `from_now`, though - maybe it's smart enough to work out either way? (or people are accidentally setting a cookie 50 years into the future 😆)

Reply:
Can we check the behaviour with something like `one second` locally? I would be more comfortable with `from_now`, cause it's obvious it will result in a set point in time, whereas `1.year` sounds like duration, and I don't know how magic of the Rails makes it work :P

Reply:
pfffft a rails dev complaining about magic 🤪 

(I'm changing this now)